internal/provider/resources/resource_deployment.go

```diff
@@ -1042,6 +1068,41 @@ func RequestScalingSpec(ctx context.Context, scalingSpecObj types.Object) (*plat
 	return platformScalingSpec, nil
 }
 
+// RequestRemoteExecution converts a Terraform object to a platform.RemoteExecutionRequest to be used in create and update requests
+func RequestRemoteExecution(ctx context.Context, remoteExecutionObj types.Object) (*platform.DeploymentRemoteExecutionRequest, diag.Diagnostics) {
```

Comment:
could you also add tests for this to the `resource_deployment_test` file? generic deployment tests and some tests on input validation would also be good (eg. validating that remote execution is only allowed on dedicated)